collect_app/src/main/java/org/odk/collect/android/activities/FirstLaunchActivity.kt

```diff
@@ -59,16 +61,25 @@ class FirstLaunchActivity : LocalizedActivity() {
                 versionInformation.versionToDisplay
             )
 
-            configureLater.addOnClickListener {
-                Analytics.log(AnalyticsEvents.TRY_DEMO)
+            dontHaveServer.apply {
+                text = SpannableStringBuilder()
+                    .append(getString(org.odk.collect.strings.R.string.dont_have_project))
+                    .append(" ")
+                    .color(getThemeAttributeValue(context, com.google.android.material.R.attr.colorAccent)) {
+                        append(getString(org.odk.collect.strings.R.string.try_demo))
+                    }
 
-                projectsRepository.save(Project.DEMO_PROJECT)
-                projectsDataService.setCurrentProject(Project.DEMO_PROJECT_ID)
+                setOnClickListener {
+                    Analytics.log(AnalyticsEvents.TRY_DEMO)
 
-                ActivityUtils.startActivityAndCloseAllOthers(
-                    this@FirstLaunchActivity,
-                    MainMenuActivity::class.java
-                )
+                    projectsRepository.save(Project.DEMO_PROJECT)
```

Comment:
This is definitely an existing problem, but while we're here it feels like it'd be a good time to move our project saving (for at least this screen) off of the UI thread. We currently store the projects in a (eventually) `SharedPreferences` backed repository so `apply` should take care of the actual IO being off the UI thread for us, but we're doing JSON parsing and serialisation in `save` which I think is probably heavy enough that we should be doing it using `Scheduler`.

Reply:
Done.